subsys/caf/modules/ble_adv.c

```diff
@@ -598,6 +674,32 @@ static void fast_adv_end_fn(struct k_work *work)
 	__ASSERT_NO_MSG(!fast_adv);
 }
 
+static void rpa_rotate_fn(struct k_work *work)
+{
+	ARG_UNUSED(work);
+
+	if (IS_ENABLED(CONFIG_CAF_BLE_ADV_GRACE_PERIOD) && (state == STATE_GRACE_PERIOD)) {
+		(void)k_work_cancel_delayable(&grace_period_end);
+		grace_period_end_fn(NULL);
+		return;
+	}
+
+	__ASSERT_NO_MSG(state == STATE_ACTIVE);
+	__ASSERT_NO_MSG(!direct_adv);
+
+	int err = bt_le_adv_stop();
+
+	if (!err) {
+		__ASSERT_NO_MSG(!req_new_adv_session);
+		force_rpa_rotation(cur_identity);
+		ble_adv_start();
```

Comment:
We could even rely on ble_adv_start_undirected, since we know that we are not advertising directly

Reply:
sure